dali/operators/image/convolution/laplacian_params.h

```diff
@@ -0,0 +1,302 @@
+    if (!has_win_size_t_input_) {
+      int prev_size = window_sizes_.size();
+      if (prev_size < nsamples) {
+        auto window_sizes = prev_size == 0 ? ObtainConstWindowSizes() : window_sizes_[0];
+        window_sizes_.resize(nsamples, window_sizes);
+      }
+    } else {
+      window_sizes_.resize(nsamples);
+      for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+        const auto &tv = ws.ArgumentInput(kWindowSizeArgName);
+        const auto &tensor = tv[sample_idx];
+        const auto &shape = tensor.shape();
+        auto vol = volume(shape);
+        DALI_ENFORCE(vol == 1 || vol == axes || vol == num_windows,
+                     make_string("Argument `", kWindowSizeArgName, "` for sample ", sample_idx,
+                                 " is expected to have 1, ", axes, " or ", axes, "x", axes,
+                                 " elements, got ", vol, "."));
+        SetSampleWindowSizes(window_sizes_[sample_idx],
+                             make_span(tensor.template data<int>(), vol));
+        ValidateWindowSizes(window_sizes_[sample_idx], sample_idx);
+      }
+    }
+  }
+
+  std::array<std::array<int, axes>, axes> ObtainConstWindowSizes() {
+    std::vector<int> tmp;
+    if (!spec_.TryGetRepeatedArgument<int>(tmp, kWindowSizeArgName)) {
+      int scalar = spec_.GetArgument<int>(kWindowSizeArgName);
+      tmp.assign(axes, scalar);
+    }
+    std::array<std::array<int, axes>, axes> window_sizes;
+    auto in_size = tmp.size();
+    DALI_ENFORCE(in_size == 1 || in_size == axes || in_size == num_windows,
+                 make_string("Argument `", kWindowSizeArgName, "` is expected to have 1, ", axes,
+                             " or ", axes, "x", axes, " elements, got ", in_size, "."));
+    SetSampleWindowSizes(window_sizes, make_span(tmp));
+    ValidateWindowSizes(window_sizes, 0);
+    return window_sizes;
+  }
+
+  void ValidateWindowSizes(const std::array<std::array<int, axes>, axes> &window_sizes,
+                         int sample_idx) {
+    for (int i = 0; i < axes; i++) {
+      for (int j = 0; j < axes; j++) {
+        bool is_deriv = i == j;
+        auto win_size = window_sizes[i][j];
+        auto min_size = is_deriv ? 3 : 1;
+        DALI_ENFORCE(
+            min_size <= win_size && win_size <= maxWindowSize && win_size % 2 == 1,
+            make_string((is_deriv ? "Derivative " : "Smoothing "),
+                        "window size must be an odd integer between ", min_size, " and ",
+                        maxWindowSize, ", got ", win_size, " for sample: ", sample_idx, "."));
+      }
+    }
+  }
+
+  void SetSampleWindowSizes(std::array<std::array<int, axes>, axes> &window_sizes,
+                            span<const int> in) {
+    auto in_size = in.size();
+    if (in_size == num_windows) {
+      for (int i = 0; i < axes; i++) {
+        memcpy(window_sizes[i].data(), in.data() + i * axes, sizeof(int) * axes);
```

Comment:
I guess it doesn't make much of the difference - can you use std::copy here?

Reply:
done